ui-v2/app/adapters/application.js

```diff
@@ -1,8 +1,46 @@
 import Adapter from './http';
 import { inject as service } from '@ember/service';
+import { get } from '@ember/object';
 
 export const DATACENTER_QUERY_PARAM = 'dc';
 export default Adapter.extend({
   repo: service('settings'),
   client: service('client/http'),
+  // TODO: kinda protected for the moment
+  // decide where this should go either read/write from http
+  // should somehow use this or vice versa
+  request: function(req, resp, obj, modelName) {
+    const client = get(this, 'client');
+    const store = get(this, 'store');
+    const adapter = this;
+
+    let unserialized, serialized;
+    const serializer = store.serializerFor(modelName);
+    // workable way to decide whether this is a snapshot
+    // essentially 'is attributable'.
+    // Snapshot is private so we can't do instanceof here
+    if (typeof obj.attributes === 'function') {
+      unserialized = obj.attributes();
+      serialized = serializer.serialize(obj, {});
+    } else {
+      unserialized = obj;
+      serialized = unserialized;
+    }
+
+    return client
+      .request(function(request) {
+        return req(adapter, request, serialized, unserialized);
+      })
+      .catch(function(e) {
+        return adapter.error(e);
+      })
+      .then(function(response) {
+        // TODO: When HTTPAdapter:responder changes, this will also need to change
+        return resp(serializer, response, serialized, unserialized);
+      });
+    // TODO: Potentially add specific serializer errors here
```

Comment:
Probably yes. I have another bunch of code for a similar Serializer refactor 😬 , I'm not ready to PR it yet though (this series of Adapter refactoring PR's gets me over the line for the ember upgrade so no need to PR the serializer stuff just yet) - this realtes to the first TODO here.

I'm also half sticking to previous `ember-data` functionality here (I'm keeping the jQuery usage for now even though I don't want to etc). Specifically here - the ember-data provided adapters/serializers have specific adapter errors, but not specific serializer errors, and I'm thinking that potentially it might be beneficial to add some here, I'm not 100% sold on that idea yet - and I don't really want to add it here, but maybe later - this relates to the second TODO here.

I wish I had a better way to hint to folks 2 different types of TODO's 😢 i.e:

1. TODO-type-1: I need to do this during this PR - but I'm looking for any suggestions/advice/information on this TODO because I'm not happy with what I've done myself, which is why I've PR'ed it with the TODO in.
2. TODO-type-2: This stuff is longer term TODO, i.e. TODO in another PR, possibly by someone else, potentially before or after going into master.

Both of the above would be TODO-type-2.

Reply:
I use `FIXME` for the first kind of todo, because I never want to commit `FIXME`s to the primary branch, but I’m fine with leaving `TODO`s hehe

Reply:
Ahh sweet of course yeah, `FIXME` has a better ring to it than `TODO-type-1` 😂 .

Will try to use that moving forwards, thanks!